File: src-backend/app/models/analysis.py
# src-backend/app/models/analysis.py
from pydantic import BaseModel


class AttentionPair(BaseModel):
    source: str
    target: str
    weight: float
    classification: str


class AttentionMatrixResponse(BaseModel):
    model_id: str
    feature_names: list[str]
    matrix: list[list[float]]
    pairs: list[AttentionPair]
    threshold: float


class HeatmapDataResponse(BaseModel):
    model_id: str
    feature_names: list[str]
    values: list[list[float]]
    min_value: float
    max_value: float


class NetworkNode(BaseModel):
    id: str
    name: str


class NetworkEdge(BaseModel):
    source: str
    target: str
    weight: float
    classification: str


class NetworkGraphResponse(BaseModel):
    model_id: str
    nodes: list[NetworkNode]
    edges: list[NetworkEdge]
    threshold: float

File: src-frontend/src/types/analysis.ts
// src/types/analysis.ts
export interface AttentionPair {
  source: string;
  target: string;
  weight: number;
  classification: "synergistic" | "antagonistic";
}

export interface AttentionMatrixResponse {
  model_id: string;
  feature_names: string[];
  matrix: number[][];
  pairs: AttentionPair[];
  threshold: number;
}

export interface HeatmapDataResponse {
  model_id: string;
  feature_names: string[];
  values: number[][];
  min_value: number;
  max_value: number;
}

export interface NetworkNode {
  id: string;
  name: string;
}

export interface NetworkEdge {
  source: string;
  target: string;
  weight: number;
  classification: "synergistic" | "antagonistic";
}

export interface NetworkGraphResponse {
  model_id: string;
  nodes: NetworkNode[];
  edges: NetworkEdge[];
  threshold: number;
}
